pkg/docker/config/config.go

```diff
@@ -216,12 +217,29 @@ func getAuthFilePaths(sys *types.SystemContext, homeDir string) []authPath {
 // file or .docker/config.json, including support for OAuth2 and IdentityToken.
 // If an entry is not found, an empty struct is returned.
 func GetCredentials(sys *types.SystemContext, registry string) (types.DockerAuthConfig, error) {
-	return getCredentialsWithHomeDir(sys, registry, homedir.Get())
+	return getCredentialsWithHomeDir(sys, nil, registry, homedir.Get())
+}
+
+// GetCredentialsWithRef returns the registry credentials stored in the
+// registry-specific credential helpers or in the default global credentials
+// helpers with falling back to using either auth.json
+// file or .docker/config.json, including support for OAuth2 and IdentityToken.
+// Compared to GetCredentials, this function accepts a full ref, which will
+// be used for supporting multiple paths when looking up the credentials in
+// auth.json. Path support means, that a given ref
+// "registry.io/some/path/image:latest" will lookup:
+// - registry.io/some/path/image
+// - registry.io/some/path
+// - registry.io/some
+// - registry.io (fallback for compatibility with GetCredentials)
```

Comment:
This should not be an API promise. Callers (unlike software _creating_ auth.json) have no need to care about this, and new features frequently enough cause us to break such promises.

(In fact I’m tempted to drop even the list of sources this function reads from, above; just say “credentials appropriate for `sys` and the users’ environment”.)

Reply:
Fixed as suggested and removed the list.